packages/components/src/components/Select/Combobox.tsx

```diff
@@ -52,77 +57,105 @@ export const SelectCombobox = (
+    } = useSelectData(children, allowCustomValue);
 
     const {
+        isOpen,
+        highlightedIndex,
+        inputValue,
+        selectedItem,
+        selectItem,
         getInputProps,
         getToggleButtonProps,
         getMenuProps,
         getItemProps,
         reset,
-        isOpen,
-        highlightedIndex,
-        inputValue,
     } = useCombobox({
         items,
+        defaultHighlightedIndex: 0,
+        selectedItem: getItemByValue(value),
+        defaultSelectedItem: getItemByValue(defaultValue),
         onSelectedItemChange: ({ selectedItem }) => {
             onSelect && onSelect(selectedItem.value);
         },
-        selectedItem: activeItem,
         onInputValueChange: ({ inputValue }) => {
             setFilterText(inputValue);
         },
-        defaultSelectedItem: defaultItem,
-        defaultHighlightedIndex: 0,
         itemToString: (item) => (item ? item.label : ''),
     });
 
     const wasClicked = useRef(false);
-    const valueInvalid = useMemo(
-        () => inputValue && !items.find((item) => item.label.toLowerCase().includes(inputValue.toLowerCase())),
-        [inputValue, items],
-    );
+
+    const onBlurHandler = (blurEvent: FocusEvent<HTMLInputElement, Element>) => {
+        blurEvent.target.dataset.showFocusRing = 'false';
+        wasClicked.current = false;
+
+        if (valueExists) {
+            if (existingValueItem?.label) {
+                // if the value exists, normalize the casing on the input
+                selectItem(existingValueItem);
+            }
+        } else {
+            const selectedItemNullOrOutdated =
+                selectedItem?.label.toLocaleLowerCase() !== inputValue.toLocaleLowerCase();
+
+            if (selectedItemNullOrOutdated) {
+                // if there is no selection or
+                // the existing selected value is not the same as the input value (old),
+                // reset the input
+                reset();
+            }
+        }
+
+        if (getInputProps().onBlur) {
+            getInputProps().onBlur?.(blurEvent);
+        }
+    };
 
     return (
         <RadixPopover.Root open={isOpen}>
             <RadixPopover.Anchor asChild>
                 <div ref={forwardedRef} className={styles.root} data-error={valueInvalid}>
                     <input
+                        data-test-id={dataTestId}
+                        {...getInputProps({
+                            'aria-label': ariaLabel,
+                        })}
+                        placeholder={placeholder}
+                        className={styles.input}
+                        disabled={disabled}
                         onMouseDown={(mouseEvent) => {
                             wasClicked.current = true;
                             mouseEvent.currentTarget.dataset.showFocusRing = 'false';
                         }}
-                        placeholder={placeholder}
-                        {...getInputProps({
```

Comment:
reordering this might cause issues with overrides that are not intended.

Reply:
will revert

Reply:
still has changed order